pkg/agent/util/net_windows.go

```diff
@@ -22,46 +22,86 @@ import (
 	"fmt"
 	"net"
 	"strings"
+	"time"
 
 	"github.com/Microsoft/go-winio"
 	"github.com/Microsoft/hcsshim"
 	ps "github.com/benmoss/go-powershell"
 	"github.com/benmoss/go-powershell/backend"
 	"github.com/containernetworking/plugins/pkg/ip"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/klog"
 )
 
 const (
-	ContainerVNICPrefix = "vEthernet"
-	HNSNetworkType      = "Transparent"
-	LocalHNSNetwork     = "antrea-hnsnetwork"
-	OVSExtensionID      = "583CC151-73EC-4A6A-8B47-578297AD7623"
-	namedPipePrefix     = `\\.\pipe\`
+	ContainerVNICPrefix  = "vEthernet"
+	HNSNetworkType       = "Transparent"
+	LocalHNSNetwork      = "antrea-hnsnetwork"
+	OVSExtensionID       = "583CC151-73EC-4A6A-8B47-578297AD7623"
+	namedPipePrefix      = `\\.\pipe\`
+	commandMaxRetry      = 5
+	commandRetryInternal = time.Second
 )
 
 func GetNSPath(containerNetNS string) (string, error) {
 	return containerNetNS, nil
 }
 
+func GetHostInterfaceStatus(ifaceName string) (string, error) {
+	cmd := fmt.Sprintf(`Get-NetAdapter -InterfaceAlias "%s" | Select-Object -Property Status | Format-Table -HideTableHeaders`, ifaceName)
+	out, err := CallPSCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(out), nil
+}
+
 // EnableHostInterface sets the specified interface status as UP.
 func EnableHostInterface(ifaceName string) error {
-	cmd := fmt.Sprintf("Enable-NetAdapter -InterfaceAlias %s", ifaceName)
-	return InvokePSCommand(cmd)
+	cmd := fmt.Sprintf(`Enable-NetAdapter -InterfaceAlias "%s"`, ifaceName)
+	// Enable-NetAdapter is not a block operation by our test. It returns immediately no matter if the interface
+	// has been enabled. So we need to check interface status to ensure the interface is up before return.
+	return wait.PollImmediate(commandRetryInternal, commandMaxRetry, func() (done bool, err error) {
```

Comment:
I think that we should wrap the error returned by `wait.PollImmediate`, either here or at the caller. Otherwise I believe we will get a very unhelpful error message in case of timeout.

Reply:
Done.